rm-main/src/ui/tabs/torrents/tasks/status.rs

```diff
@@ -0,0 +1,148 @@
+                            format!(" Error adding {display_name}")
+                        }
+                        StatusTask::Delete(name) => {
+                            let display_name = format_display_name(&name);
+                            format!(" Error deleting {display_name}")
+                        }
+                    },
+                    CurrentTaskState::Success(_) => match &self.task {
+                        StatusTask::Add(name) => {
+                            let display_name = format_display_name(&name);
+                            format!(" Added {display_name}")
+                        }
+                        StatusTask::Delete(name) => {
+                            let display_name = format_display_name(&name);
+                            format!(" Deleted {display_name}")
+                        }
+                    },
+                    _ => return,
+                };
+                let mut line = Line::default();
+                match task_state {
+                    CurrentTaskState::Failure => {
+                        line.push_span(Span::styled("", Style::default().red()));
+                    }
+                    CurrentTaskState::Success(_) => {
+                        line.push_span(Span::styled("", Style::default().green()));
+                    }
+                    _ => return,
+                }
+
+                line.push_span(Span::raw(status_text));
+                let paragraph = Paragraph::new(line);
+                f.render_widget(paragraph, rect);
+            }
+        }
+    }
+
+    fn handle_actions(&mut self, action: Action) -> Option<Action> {
+        match action {
+            Action::Tick => self.tick(),
+            Action::TaskSuccess => {
+                self.task_status.success(tokio::time::Instant::now());
+                Some(Action::Render)
+            }
+            Action::Error(_) => {
+                self.task_status.failure();
+                Some(Action::Render)
+            }
+            _ => Some(action),
+        }
+    }
+
+    fn tick(&mut self) -> Option<Action> {
+        match &self.task_status {
+            CurrentTaskState::Loading(state) => {
+                state.lock().unwrap().calc_next();
+                Some(Action::Render)
+            }
+            CurrentTaskState::Success(start) => {
+                let expiration_duration = tokio::time::Duration::from_secs(5);
+                if start.elapsed() > expiration_duration {
+                    return Some(Action::Quit);
+                }
+                None
+            }
+            _ => None,
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum StatusTask {
+    Add(String),
+    Delete(String),
+}
+
+#[derive(Clone)]
+pub enum CurrentTaskState {
+    Loading(Arc<Mutex<ThrobberState>>),
+    Success(Instant),
+    Failure,
+}
+
+impl CurrentTaskState {
+    fn failure(&mut self) {
```

Comment:
set_failure and set_success I think would be more readable when reading the code